### PR TITLE
NoUnreachableDefaultArgumentValueFixer - Mark as risky

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2561

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -426,8 +426,9 @@ Choose from the list of available rules:
    | *Rule is: configurable.*
 
 * **no_unreachable_default_argument_value**
-   | In method arguments there must not be arguments with default values
-   | before non-default ones.
+   | In function and method arguments there must not be arguments with
```
Comment (keradus) [inline README.rst]: method is a subtype of function

File: src/Fixer/FunctionNotation/NoUnreachableDefaultArgumentValueFixer.php [hunk 2]
```diff
@@ -53,18 +48,38 @@ public function fix(\SplFileInfo $file, Tokens $tokens)
...
                     '<?php
 function example($foo = "two words", $bar) {}
 '
                 ),
-            )
+            ),
+            null,
+            null,
+            null,
+            'Modifies the signature of methods/functions, risky when using systems relying on those (for example through reflection).'
```
Comment (keradus) [inline src/Fixer/FunctionNotation/NoUnreachableDefaultArgumentValueFixer.php]: could we ref. to example from Symfony, as the statement is not obvious for all devs.

File: README.rst (hunk 1 above)
Comment (SpacePossum) [inline README.rst]: indeed, so you want `function` only here?

File: src/Fixer/FunctionNotation/NoUnreachableDefaultArgumentValueFixer.php (hunk 2 above)
Comment (SpacePossum) [inline src/Fixer/FunctionNotation/NoUnreachableDefaultArgumentValueFixer.php]: not sure SF does this, do you made have an example? (I don't use SF that much, mainly Silex)

Comment (keradus) [inline src/Fixer/FunctionNotation/NoUnreachableDefaultArgumentValueFixer.php]: initial post of https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2485
also, when you process input parameters for API routes.

ok, let as leave it as is then ;)

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**